lib/engine/game/g_1822_pnw/step/acquire_company.rb

```diff
@@ -95,21 +95,21 @@ def process_choose(action)
           end
 
           def p20_targets
-            bidbox_corporations = @game.bidbox_minors.map { |c| @game.corporation_from_company(c) }
-            targets = @game.corporations.select { |c| @game.associated_minor?(c) && !c.owner && !bidbox_corporations.include?(c) }
+            bidbox_corporations = ::Set.new(@game.bidbox_minors) { |c| @game.corporation_from_company(c) }
 
-            # include minors that fell through the bidboxes without bids
-            @game.minor_associations.each do |minor_id, major_id|
+            @game.minor_associations.each_with_object([]) do |(minor_id, major_id), targets|
               company = @game.company_by_id("M#{minor_id}")
               minor = @game.corporation_by_id(minor_id)
               major = @game.corporation_by_id(major_id)
 
-              targets << minor if !company.owner && company.closed? &&
-                                  !minor.owner && minor.closed? &&
-                                  !major.owner && !major.floated?
-            end
+              already_discarded = (!company.owner && company.closed? && minor.closed?)
+
+              minor_available = !minor.owner &&
+                                (already_discarded || !bidbox_corporations.include?(minor))
```

Comment:
Would this give the same result? It would avoid the need to create the `bidbox_corporations` set.
```suggestion
              minor_available = !minor.owner &&
                                (already_discarded || !@game.bidbox_minors.include?(company))
```

Reply:
Done in 61e8975a51d6b5f708c47991cf3f830ca140482a.

`@game.bidbox_minors` recomputes the minors every time, so I kept the call to it outside of the loop here.